Snail/Replay.swift

```diff
@@ -0,0 +1,35 @@
+//  Copyright © 2016 Compass. All rights reserved.
+
+import Foundation
+
+public class Replay<T>: Observable<T> {
+    private let threshold: Int
+    private var events: [Event<E>] = []
+
+    public init(_ threshold: Int) {
+        self.threshold = threshold
+    }
+
+    public override func subscribe(_ handler: @escaping (Event<E>) -> Void) {
+        super.subscribe(handler)
+        replay(handler)
+    }
+
+    public override func subscribe(onNext: ((T) -> Void)? = nil, onError: ((Error) -> Void)? = nil, onDone: (() -> Void)? = nil) {
+        super.subscribe(onNext: onNext, onError: onError, onDone: onDone)
+        replay(createHandler(onNext: onNext, onError: onError, onDone: onDone))
+    }
+
+    public override func on(_ event: Event<E>) {
+        events.append(event)
```

Comment:
I think we might want to do something like append, then reset the array to be the threshold, so it doesn't just grow forever

```swift
events.append(event)
events = events.suffix(threshold)
```